internal/controller/eventing/integrationtests/validation/integration_test.go

```diff
@@ -0,0 +1,683 @@
+		panic(err)
+	}
+
+	os.Exit(code)
+}
+
+// Test_Validate_CreateEventing creates an eventing CR with correct and purposefully incorrect values, and compares
+// the error that was caused by this against a wantErrMsg to test the eventing CR validation rules.
+func Test_Validate_CreateEventing(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name                      string
+		givenUnstructuredEventing unstructured.Unstructured
+		wantErrMsg                string
+	}{
+		{
+			name: `validation of spec.publisher.replicas.min fails for values > max`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						publisher: map[string]any{
+							replicas: map[string]any{
+								min: 3,
+								max: 2,
+							},
+						},
+					},
+				},
+			},
+			wantErrMsg: "min value must be smaller than the max value",
+		},
+		{
+			name: `validation of spec.publisher.replicas.min passes for values <= max`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						publisher: map[string]any{
+							replicas: map[string]any{
+								min: 2,
+								max: 2,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.backends.config.eventMeshSecret fails when empty if spec.type = EventMesh`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						backends: map[string]any{
+							backendType: typeEventMesh,
+							config:      map[string]any{},
+						},
+					},
+				},
+			},
+			wantErrMsg: "secret cannot be empty if EventMesh backend is used",
+		},
+		{
+			name: `validation of spec.backends.config.eventMeshSecret passes when set if spec.type = EventMesh`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						backends: map[string]any{
+							backendType: typeEventMesh,
+							config: map[string]any{
+								eventMeshSecret: someSecret,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.backends.config.eventMeshSecret passes when empty if spec.type = NATS`,
```

Comment:
Looks like a duplicate of the test case `validation of spec.backends.type passes for value = NATS`

Reply:
Removed